core/evm.go

```diff
@@ -27,18 +27,62 @@
 package core
 
 import (
+	"bytes"
 	"math/big"
 
 	"github.com/ava-labs/coreth/consensus"
 	"github.com/ava-labs/coreth/consensus/misc/eip4844"
+	"github.com/ava-labs/coreth/core/extstate"
+	"github.com/ava-labs/coreth/core/state"
 	"github.com/ava-labs/coreth/core/types"
-	"github.com/ava-labs/coreth/core/vm"
+	"github.com/ava-labs/coreth/params"
 	"github.com/ava-labs/coreth/predicate"
 	"github.com/ethereum/go-ethereum/common"
+	gethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/core/vm"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/holiman/uint256"
 )
 
+func init() {
+	vm.RegisterHooks(hooks{})
+}
+
+type hooks struct{}
+
+func (hooks) OverrideNewEVMArgs(args *vm.NewEVMArgs) *vm.NewEVMArgs {
+	rules := args.ChainConfig.Rules(args.BlockContext.BlockNumber, args.BlockContext.Random != nil, args.BlockContext.Time)
+	args.StateDB = wrapStateDB(rules, args.StateDB)
+
+	if params.GetRulesExtra(rules).IsDurango {
+		args.BlockContext.Random = new(common.Hash)
+		args.BlockContext.Random.SetBytes(args.BlockContext.Difficulty.Bytes())
```

Comment:
It might break something, but also not setting it may break things as Random != nil is often used to determine if the merge has occurred. It is something we will have to be careful about with testing.

Unfortunately some blocks also expect .Difficulty to be nil after merge, so this can break something as well (where both are set)

Reply:
shall we rather use Etna upgrade then for this (Cancun)?

Reply:
this is activated in Merge so it's prior to Shanghai in upstream